sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -68,9 +68,9 @@ case class SortMergeJoinExec(
       val leftKeyOrdering = getKeyOrdering(leftKeys, left.outputOrdering)
       val rightKeyOrdering = getKeyOrdering(rightKeys, right.outputOrdering)
       leftKeyOrdering.zip(rightKeyOrdering).map { case (lKey, rKey) =>
-        // Also add the right key and its `sameOrderExpressions`
-        SortOrder(lKey.child, Ascending, lKey.sameOrderExpressions + rKey.child ++ rKey
-          .sameOrderExpressions)
+        // Also add expressions from right side sort order
+        val sameOrderExpressions = ExpressionSet(lKey.children ++ rKey.children) - lKey.child
```

Comment:
why not `ExpressionSet(lKey.sameOrderExpressions ++ rKey.children)`?

Reply:
I was thinking if "lKey.child" can be part of rKey.children. But that shouldn't be the case as left and right side of SMJ will always return different expressions ids (even when they work on same tables).

making suggested change here.